src/main/scala/karazin/scala/users/group/week1/homework/Homework.scala

```diff
@@ -56,74 +57,74 @@ def or(left: Boolean, right: Boolean): Boolean =
 
 end `Boolean Operators`
 
-object `Fermat Numbers` :
+object `Fermat Numbers`:
 
   val multiplication: (BigInt, BigInt) => BigInt =
-    val multiplicationRec:(BigInt,BigInt,BigInt) =>BigInt=
-      (a,b,res)=>
-        if b<1 then res
-        else multiplicationRec(a,b-1,res+a)
-    (a, b)=>
-      if b > 0 then multiplicationRec(a,b,0)
-      else -1*multiplicationRec(a,b.abs,0)
+    val multiplicationRec: (BigInt, BigInt, BigInt) => BigInt =
+      (a, b, res) =>
+        if b < 1 then res
+        else multiplicationRec(a, b - 1, res + a)
+    (a, b) =>
+      if b > 0 then multiplicationRec(a, b, 0)
+      else -1 * multiplicationRec(a, b.abs, 0)
 
   //assume that the exponent is a positive number
   val power: (BigInt, BigInt) => BigInt =
-    val powerRec:(BigInt,BigInt,BigInt)=> BigInt=
-      (a,b,res)=>
-        if b<1 then res
-        else powerRec(a,b,multiplication(a,res))
-    (a,b)=>
-      powerRec(a,b,1)
+    val powerRec: (BigInt, BigInt, BigInt) => BigInt =
+      (a, b, res) =>
+        if b < 1 then res
+        else powerRec(a, b, multiplication(a, res))
+    (a, b) =>
+      powerRec(a, b, 1)
 
   val fermatNumber: Int => BigInt =
-    n=>
-      power(2,power(2,n))+1
+    n =>
+      power(2, power(2, n)) + 1
 
 end `Fermat Numbers`
 
 //extremely unsure about this function
 //added recursion instead of var
-object `Look-and-say Sequence` :
+object `Look-and-say Sequence`:
   val lookAndSaySequenceElement: Int => Array[Int] =
```

Comment:
Array is a mutable structure. We agreed from the beginning that mutable structures are forbidden.